pkg/admission/certwriter/certwriter.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certwriter
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kubernetes-sigs/controller-runtime/pkg/admission/certgenerator"
+	"github.com/kubernetes-sigs/controller-runtime/pkg/client"
+)
+
+const (
+	// CACertName is the name of the CA certificate
+	CACertName = "ca-cert.pem"
+	// ServerKeyName is the name of the server private key
+	ServerKeyName = "key.pem"
+	// ServerCertName is the name of the serving certificate
+	ServerCertName = "cert.pem"
+)
+
+// Options are options for configuring a CertOutput.
+type Options struct {
+	Client        client.Client
+	CertGenerator certgenerator.CertGenerator
+}
+
+// CertWriter provides method to handle webhooks.
+type CertWriter interface {
+	// Create a new CertWriter with the given runtime.Object
+	New(runtime.Object) (CertWriter, error)
```

Comment:
Are we sure this method belongs in this interface ? its more like a constructor for CertWriter itself ?